python/lsst/cp/testing/isr.py

```diff
@@ -0,0 +1,302 @@
+        super().__init__(config=config)
+
+        if not config.doBias:
+            del self.bias
+        if not config.doLinearize:
+            del self.linearizer
+        if not config.doCrosstalk:
+            del self.crosstalkSources
+            del self.crosstalk
+        if not config.doBrighterFatter:
+            del self.bfKernel
+            del self.newBFKernel
+        if not config.doDefect:
+            del self.defects
+        if not config.doDark:
+            del self.dark
+        if not config.doFlat:
+            del self.flat
+        if not config.doFringe:
+            del self.fringes
+        if not config.doStrayLight:
+            del self.strayLightData
+        if not config.usePtcGains and not config.usePtcReadNoise:
+            del self.ptc
+        if not config.doAttachTransmissionCurve:
+            del self.opticsTransmission
+            del self.filterTransmission
+            del self.sensorTransmission
+            del self.atmosphereTransmission
+        else:
+            if not config.doUseOpticsTransmission:
+                del self.opticsTransmission
+            if not config.doUseFilterTransmission:
+                del self.filterTransmission
+            if not config.doUseSensorTransmission:
+                del self.sensorTransmission
+            if not config.doUseAtmosphereTransmission:
+                del self.atmosphereTransmission
+        if not config.doIlluminationCorrection:
+            del self.illumMaskedImage
+
+        if not config.doWrite:
+            del self.outputExposure
+            del self.preInterpExposure
+            del self.outputFlattenedThumbnail
+            del self.outputOssThumbnail
+        if not config.doSaveInterpPixels:
+            if "preInterpExposure" in self.outputs:
+                del self.preInterpExposure
+        if not config.qa.doThumbnailOss:
+            if "outputOssThumbnail" in self.outputs:
+                del self.outputOssThumbnail
+        if not config.qa.doThumbnailFlattened:
+            if "outputFlattenedThumbnail" in self.outputs:
+                del self.outputFlattenedThumbnail
+
+
+class CptIsrTaskConfig(IsrTaskConfig,
+                       pipelineConnections=CptIsrTaskConnections):
+    """Alternate no-prerequisite input ISR Config.
+
+    Regular processing should not use this task.
+    """
+
+    expectedExposureType = pexConfig.Field(
+        dtype=str,
+        doc="Type of exposures that should be processed.",
+        default="",
+    )
+    expectedObservationReason = pexConfig.Field(
+        dtype=str,
+        doc="Further restrict processed exposures by observation_reason.",
+        default="",
+    )
+
+
+class CptIsrTask(IsrTask):
+    """Alternate no-prerequisite input ISR Task.
+
+    Regular processing should not use this task.
+    """
+
+    ConfigClass = CptIsrTaskConfig
+    _DefaultName = "cptIsrTask"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
```

Comment:
I'm confused ... didn't we need this to be an adjustQuantum to filter out early?  I must be missing something.